### PR TITLE
Zarr append for datasets (non-reference)

### PR DESCRIPTION
**Pull request by mavaylon1**
## Motivation

What was the reasoning behind this change? Please explain the changes briefly.

This supports appending to zarr arrays. This does not cover the edge case for dataset of references. This has the base tests for append_data. There are more tests (TBD) in hdmf-zarr that incorporate the full IO process. 

## How to test the behavior?
```
Show how to reproduce the new behavior (can be a bug fix or a new feature)
```

## Checklist

- [x] Did you update `CHANGELOG.md` with your changes?
- [x] Does the PR clearly describe the problem and the solution?
- [x] Have you reviewed our [Contributing Guide](https://github.com/hdmf-dev/hdmf/blob/dev/docs/CONTRIBUTING.rst)?
- [x] Does the PR use "Fix #XXX" notation to tell GitHub to close the relevant issue numbered XXX when the PR is merged?

### CODE REVIEW

File: src/hdmf/validate/validator.py [hunk 1]
```diff
@@ -164,7 +164,7 @@ def get_type(data, builder_dtype=None):
                 # Empty array
                 else:
                     # Empty string array
-                    if data.dtype.metadata["vlen"] == str:
+                    if isinstance(data.dtype.metadata["vlen"], str):
```
Comment (stephprince) [inline src/hdmf/validate/validator.py]:
```suggestion
                    if data.dtype.metadata["vlen"] is str:
```

I think the original code here was checking if it is the str class itself

**Review by stephprince — CHANGES_REQUESTED** [review]
small suggestion otherwise looks good to me!

**Review by stephprince — APPROVED** [review]
